database-commons/src/main/java/io/cdap/plugin/db/batch/source/StatementFunction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.batch.source;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * Used for {@link StatementModifyingConnection} to represent a function which can be applied to a statement.
+ * @param <T>
```

Comment:
nit: should have a non-empty javadoc "type of statement to modify" or something like that